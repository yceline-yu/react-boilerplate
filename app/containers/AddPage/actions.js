/*
 * AddPage Actions
 *
 */

import { CHANGE_STRING, SUBMIT_STRING, SUBMIT_ERROR } from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {string} string The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_STRING
 */
export function changeString(string) {
  return {
    type: CHANGE_STRING,
    string,
  };
}

/**
 * Changes status of the form
 *
 * @param  {string} submit The final version of the input field
 *
 * @return {object} An action object with a type of SUBMIT_STRING
 */
export function submitString(submit) {
  return {
    type: SUBMIT_STRING,
    submit,
  };
}

/**
 * Dispatched when posting fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of SUBMIT_ERROR passing the error
 */
export function submitError(error) {
  return {
    type: SUBMIT_ERROR,
    error,
  };
}
