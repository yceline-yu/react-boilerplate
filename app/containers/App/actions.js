/*
 * App Actions
 *
 */

import {
  LOAD_STRINGS,
  LOAD_STRINGS_SUCCESS,
  LOAD_STRINGS_ERROR,
} from './constants';

/**
 * Load the strings, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_STRINGS
 */
export function loadStrings() {
  return {
    type: LOAD_STRINGS,
  };
}

/**
 * Dispatched when the strings are loaded by the request saga
 *
 * @param  {object} strings The strings from server
 * @return {object} An action object with a type of LOAD_STRINGS_SUCCESS passing the strings
 */
export function stringsLoaded(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}

/**
 * Dispatched when loading the strings fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of LOAD_STRINGS_ERROR passing the error
 */
export function stringsLoadingError(error) {
  return {
    type: LOAD_STRINGS_ERROR,
    error,
  };
}
