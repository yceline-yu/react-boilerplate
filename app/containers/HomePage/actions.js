// /*
//  * HomePage Actions
//  *
//  * Actions change things in your application
//  * Since this boilerplate uses a uni-directional data flow, specifically redux,
//  * we have these actions which are the only way your application interacts with
//  * your application state. This guarantees that your state is up to date and nobody
//  * messes it up weirdly somewhere.
//  *
//  * To add a new Action:
//  * 1) Import your constant
//  * 2) Add a function like this:
//  *    export function yourAction(var) {
//  *        return { type: YOUR_ACTION_CONSTANT, var: var }
//  *    }
//  */

// import {
//   LOAD_STRINGS,
//   LOAD_STRINGS_SUCCESS,
//   LOAD_STRINGS_ERROR,
// } from './constants';

// /**
//  * Load the strings, this action starts the request saga
//  *
//  * @return {object} An action object with a type of LOAD_STRINGS
//  */
// export function loadStrings() {
//   return {
//     type: LOAD_STRINGS,
//   };
// }

// /**
//  * Dispatched when the strings are loaded by the request saga
//  *
//  *
//  * @return {object}      An action object with a type of LOAD_STRINGS_SUCCESS passing the strings
//  */
// export function stringsLoaded(strings) {
//   return {
//     type: LOAD_STRINGS_SUCCESS,
//     strings,
//   };
// }

// /**
//  * Dispatched when loading the strings fails
//  *
//  * @param  {object} error The error
//  *
//  * @return {object}       An action object with a type of LOAD_STRINGS_ERROR passing the error
//  */
// export function stringsLoadingError(error) {
//   return {
//     type: LOAD_STRINGS_ERROR,
//     error,
//   };
// }
