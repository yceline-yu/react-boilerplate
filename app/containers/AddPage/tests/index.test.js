/**
 * Test the AddPage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { AddPage, mapDispatchToProps } from '../index';
import { changeString } from '../actions';
import configureStore from '../../../configureStore';

describe('<AddPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const string = '';
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddPage
            loading={false}
            error={false}
            string={string}
            onChangeString={() => {}}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should not call onSubmitForm if string is empty', () => {
    const submitSpy = jest.fn();
    const string = '';
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddPage
            onChangeString={() => {}}
            onSubmitForm={submitSpy}
            string={string}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('onChangeString', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeString).toBeDefined();
      });

      it('should dispatch changeString when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const fixture = 'test';
        result.onChangeString({ target: { value: fixture } });
        expect(dispatch).toHaveBeenCalledWith(changeString(fixture));
      });
    });

    describe('onSubmitForm', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onSubmitForm).toBeDefined();
      });

      // it('should dispatch submitString when called', () => {
      //   const dispatch = jest.fn();
      //   const result = mapDispatchToProps(dispatch);
      //   result.onSubmitForm();
      //   expect(dispatch).toHaveBeenCalledWith(submitString());
      // });

      // it('should preventDefault if called with event', () => {
      //   const preventDefault = jest.fn();
      //   const result = mapDispatchToProps(() => {});
      //   const evt = { preventDefault };
      //   result.onSubmitForm(evt);
      //   expect(preventDefault).toHaveBeenCalledWith();
      // });
    });
  });
});
