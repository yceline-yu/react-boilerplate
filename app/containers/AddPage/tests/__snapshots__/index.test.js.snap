// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPage /> should render and match the snapshot 1`] = `
<section
  class="Section-sc-1td0dkz-0 gOHFWS"
>
  <section
    class="Section-sc-1td0dkz-0 CenteredSection-sc-9ounr7-0 ewtOfN"
  >
    <form
      class="Form-ye6mxs-0 eelZyG"
    >
      <h2>
        New String Form
      </h2>
      <label>
        New String:
        <input
          class="Input-sc-1qwpjmw-0 dzgcPp"
          id="string"
          name="string"
          type="text"
          value=""
        />
      </label>
      <button
        class="Button-sc-1e442dj-0 jqXTfo"
        color="teal"
        disabled=""
        type="submit"
      >
        Add
      </button>
      <a
        href="/"
      >
        <button
          class="Button-sc-1e442dj-0 eaWiGN"
          color="black"
          type="button"
        >
          Back
        </button>
      </a>
    </form>
  </section>
</section>
`;
